spopt/locate/p_median.py

```diff
@@ -246,10 +253,50 @@ def from_cost_matrix(
         )
 
         if predefined_facilities_arr is not None:
-            FacilityModelBuilder.add_predefined_facility_constraint(
-                p_median, predefined_facilities_arr
+            if fulfill_predefined_fac and facility_capacities is not None:
+                sum_predefined_fac_cap = np.sum(
+                    facility_capacities[predefined_facilities_arr]
+                )
+                if sum_predefined_fac_cap <= weights.sum():
+                    FacilityModelBuilder.add_predefined_facility_constraint(
+                        p_median,
+                        predefined_facilities_arr,
+                        weights,
+                        facility_capacities,
+                    )
+                else:
+                    raise SpecificationError(
```

Comment:
Nice job on the test checking that the `SpecificationError` is being raised appropriately. In the future we will want to test that all the new exceptions are raised when appropriate, but this is a great start!